modules/openapi-generator/src/main/resources/csharp/auth/OAuthAuthenticator.mustache

```diff
@@ -81,7 +81,14 @@ namespace {{packageName}}.Client.Auth
                 .AddParameter("client_id", _clientId)
                 .AddParameter("client_secret", _clientSecret);
             var response = await client.PostAsync<TokenResponse>(request).ConfigureAwait(false);
-            return $"{response.TokenType} {response.AccessToken}";
+            
+            switch (response.TokenType?.ToLower())
```

Comment:
thanks for the fix. can you please add a line or 2 (e.g. the reason mentioned in the PR description) explaining why this fix is needed?

Reply:
Sure, it's covered in #14826 but in summary this change relates to .NET API clients for APIs using the OAuth2 client credential flow.

When using OAuth2 as the security scheme, the client code requests an access token from the token endpoint and the server returns a response which includes a 'TokenType' property along with the 'AccessToken'.  The existing implementation constructs the Authorization header for the request to the secured API by concatenating token type and access token.

The format of the Access Token Response is defined in [RFC6749](https://www.rfc-editor.org/rfc/rfc6749#section-4.2.2) where the token_type property is defined as case insensitive.  The format of a Bearer authorization header is defined in [RFC6750](https://www.rfc-editor.org/rfc/rfc6750.html#section-2.1) where 'Bearer' is shown with a capital B.

So, the token server can return 'bearer' or 'Bearer' whereas the secured API should only receive 'Bearer'.  Consequently if the token server returns 'bearer' as the token_type but the secured API adheres strictly to RFC6750 then the Authorization header will be rejected leading to a 401 response.

This change just corrects that so that if the token response has 'bearer' as the token type the Authorization header for the API request is changed to 'Bearer'.

Reply:
to be clear, I mean adding the explanation to the code as comment so that others know why this code block is added.